e107_plugins/forum/shortcodes/batch/view_shortcodes.php

```diff
@@ -1385,8 +1404,15 @@ function sc_quickreply()
 //----	}
 		}
 	}
+	function sc_qr_sbutton($parms = null)
+	{
+		global $thread;
 
-}
-
-
+		return "<input type='submit' data-token='".e_TOKEN."' data-forum-insert='".(($thread->pages == $thread->page || $thread->pages == 0) ? 1 : 0)."' data-forum-post='".$this->var['thread_forum_id']."' data-forum-thread='".$this->var['thread_id']."' data-forum-action='quickreply' name='reply' value='".($parms['value']??LAN_FORUM_2007)."' class='btn btn-success button'>";
+	}
 
+	function sc_qr_hidden($parms = null)
```

Comment:
You can place the hidden field inside the button shortcode.

Reply:
I don't think is good idea.
I've seen forms with the hidden fields in odd places inside the form.
This way we can have some flexibility... I think...

Reply:
Indeed, it's better to place it before </form>.

Reply:
Force inject donne, in case it's missing from the designer template....

I think it's all covered now...

Let me know if you want any other change...